src/v/storage/tests/batch_cache_test.cc

```diff
@@ -312,3 +314,60 @@ FIXTURE_TEST(test_random_batch_sizes, batch_cache_test_fixture) {
         BOOST_REQUIRE_LE(r.waste(), max_waste);
     }
 }
+
+FIXTURE_TEST(test_mark_clean_empty, batch_cache_test_fixture) {
+    storage::batch_cache_index index(cache);
+    index.mark_clean(model::offset(1000));
+}
+
+FIXTURE_TEST(test_mark_clean_edge, batch_cache_test_fixture) {
+    storage::batch_cache_index index(cache);
+
+    // Small batches which will go in the same range.
+    index.put(make_batch(5, model::offset(20)), is_dirty_entry::no);
+    index.put(make_batch(5, model::offset(30)), is_dirty_entry::yes);
+    index.put(make_batch(5, model::offset(40)), is_dirty_entry::yes);
+    index.put(make_batch(5, model::offset(10)), is_dirty_entry::no);
+
+    // Reclaim will not happen due to a dirty batch in the range.
+    cache.clear();
+    BOOST_CHECK(index.get(model::offset(20)).has_value());
+    BOOST_CHECK(index.get(model::offset(30)).has_value());
+    BOOST_CHECK(index.get(model::offset(40)).has_value());
+    BOOST_CHECK(index.get(model::offset(10)).has_value());
+
+    // No reclaim as there is a dirty batch with a higher offset.
+    index.mark_clean(model::offset(43));
+    BOOST_CHECK(index.get(model::offset(20)).has_value());
+    BOOST_CHECK(index.get(model::offset(30)).has_value());
+    BOOST_CHECK(index.get(model::offset(40)).has_value());
+    BOOST_CHECK(index.get(model::offset(10)).has_value());
+
+    // Reclaim will happen here and evict the whole range.
+    index.mark_clean(model::offset(44));
+    cache.clear();
+    BOOST_CHECK(!index.get(model::offset(20)));
+    BOOST_CHECK(!index.get(model::offset(30)));
+    BOOST_CHECK(!index.get(model::offset(40)));
+    BOOST_CHECK(!index.get(model::offset(10)));
+}
+
+FIXTURE_TEST(test_mark_clean_valid_iterating, batch_cache_test_fixture) {
+    storage::batch_cache_index index(cache);
+
+    index.put(make_batch(1, model::offset(1)), is_dirty_entry::no);
+    index.put(make_batch(1, model::offset(2)), is_dirty_entry::no);
+    index.put(make_batch(10, model::offset(10)), is_dirty_entry::yes);
+
+    index.mark_clean(model::offset(1));
+    index.mark_clean(model::offset(19));
+}
+
+FIXTURE_TEST(test_add_dirty_seq, batch_cache_test_fixture) {
+    storage::batch_cache_index index(cache);
+
+    index.put(make_batch(10, model::offset(10)), is_dirty_entry::yes);
+    index.put(make_batch(10, model::offset(19)), is_dirty_entry::yes);
+
+    index.mark_clean(model::offset(28));
```

Comment:
I'm kind of confused about what these are validating given there are no test conditions being checked. Should there be some checks that the cache is clean?

Reply:
Added a comment why the test exists and extra assertions (the pass meant that we didn't hit any vasserts).